File: pset1/greedy.c
#include<stdio.h>
#include<math.h>
#include<cs50.h>
int main(void)
{
    float f;int coins=0;
    printf("O hai! How much change is owed?\n");
    f=get_float();
    while(f<0)
    {
        printf("How much change is owed?\n");
        f=get_float();
    }
    int amount=round(f*100);
    while(amount>0)
    {
       if(amount>=25)
       {
           amount-=25;coins++;
       }
       else if(amount>=10)
       {
           amount-=10;coins++;
       }
       else if(amount>=5)
       {
           amount-=5;coins++;
       }
       else if(amount>=1)
       {
           amount-=1;coins++;
       }
    }
    printf("%d\n",coins);
}

File: pset6/greedy.py
import cs50

def main():
    while True:
        print("O hai! How much change is owed?")
        amount = cs50.get_float()
        if amount >= 0:
            break
    
    numberofcoins = 0
    cents = int(round(amount * 100))
    
    numberofcoins += cents // 25
    cents %= 25
    
    numberofcoins += cents // 10
    cents %= 10
    
    numberofcoins += cents // 5
    cents %= 5
    
    numberofcoins += cents
    
    print("{}".format(numberofcoins))
    
if __name__ == "__main__":
    main()